ci/prep_cython_cache.sh

```diff
@@ -1,43 +1,73 @@
 #!/bin/bash
 
 ls "$HOME/.cache/"
+
+PYX_CACHE_DIR="$HOME/.cache/pyxfiles"
+pyx_file_list=`find ${TRAVIS_BUILD_DIR} -name "*.pyx"`
+pyx_cache_file_list=`find ${PYX_CACHE_DIR} -name "*.pyx"`
+
 CACHE_File="$HOME/.cache/cython_files.tar"
 
+# Clear the cython cache 0 = NO, 1 = YES
 clear_cache=0
+
+pyx_files=`echo "$pyx_file_list" | wc -l`
+pyx_cache_files=`echo "$pyx_cache_file_list" | wc -l`
+
+if [[ pyx_files -ne pyx_cache_files ]]
+then
+        echo "Different number of pyx files"
+        clear_cache=1
+fi
+
 home_dir=$(pwd)
 
-if [ -f "$CACHE_File" ] && [ "$USE_CACHE" ]; then
+if [ -f "$CACHE_File" ] && [ "$USE_CACHE" ] && [ -d "$PYX_CACHE_DIR" ]; then
+
+    echo "Cache available - checking pyx diff"
+
+    for i in ${pyx_file_list}
+    do
+            diff=`diff -u $i $PYX_CACHE_DIR${i}`
+            if [[ $? -eq 2 ]]
+            then
+                    echo "${i##*/} can't be diffed; probably not in cache"
+                    clear_cache=1
+            fi
+            if [[ ! -z $diff ]]
+            then
+                    echo "${i##*/} has changed:"
+                    echo $diff
+                    clear_cache=1
+            fi
+    done
 
-    echo "Cache available"
-    clear_cache=1
-    # did the last commit change cython files?
-    # go back 2 commits
     if [ "$TRAVIS_PULL_REQUEST" == "false" ]
     then
-        echo "Not a PR: checking for cython files changes from last 2 commits"
-        git diff HEAD~2 --numstat | grep -E "pyx|pxd"
-        retval=$(git diff HEAD~2 --numstat | grep -E "pyx|pxd"| wc -l)
+        echo "Not a PR"
+        # Uncomment next 2 lines to turn off cython caching not in a PR
```

Comment:
I was thinking about it; as a quick way of turning off caching of cython files in the future if something goes wrong. But happy to take them out also.